pkg/traversal/reporter.go

```diff
@@ -0,0 +1,86 @@
+package traversal
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/openshift/must-gather-clean/pkg/obfuscator"
+	"gopkg.in/yaml.v3"
+	"k8s.io/klog/v2"
+)
+
+type Report struct {
+	Replacements []map[string]string `yaml:"replacements,omitempty"`
+	Omissions    []string            `yaml:"omissions,omitempty"`
+}
+
+type Reporter interface {
+	// WriteReport writes the final report into the given path, will create folders if necessary.
+	WriteReport(path string) error
+
+	// ReportOmission saves the given path as omitted to later report as such.
+	ReportOmission(path string)
+
+	// ReportObfuscators will call the Report method on all obfuscators and collect the individual obfuscation results for final reporting.
+	ReportObfuscators(obfuscators []obfuscator.Obfuscator)
+}
+
+type SimpleReporter struct {
+	lock         *sync.RWMutex
```

Comment:
see how this does but one thing to keep in mind with concurrent code is that you can often avoid locks altogether by using immutable objects.  Ie. an obfuscator returns an immutable report result which you then add to your private aggregate.  Same for omissions.

If the overhead of locking is noticeable that might be something to give a go.

Reply:
it's good to leave a comment on what the lock is intended to guard.  It looks like this case it's that the entire type, like an intrinsic lock in java

Reply:
named the lock now after what it's guarding in the Reporters